### PR TITLE
Concept tagging

### PR DESCRIPTION
**Pull request by d471061c**

### CODE REVIEW

File: frontend/src/dialogs/concept/TagSelector.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React, { useRef, useState, useEffect } from 'react'
```
Comment (tulir) [inline frontend/src/dialogs/concept/TagSelector.js]: Delete file

File: backend/schema/datamodel.prisma [hunk 2]
```diff
@@ -123,6 +133,7 @@ type Concept {
   courses: [Course!]! @relation(name: "Courses")
   resources: [Resource!]!
   workspace: Workspace! @relation(link: INLINE, name:"WorkspaceConcepts")
+  bloomsTag: Tag @default(value: REMEMBER)
```
[the remark this answers is not in the document]
Comment (d471061c) [reply backend/schema/datamodel.prisma]: Other tags will be set separately for now.

File: frontend/src/dialogs/concept/useCreateConceptDialog.js [hunk 3]
```diff
@@ -22,11 +23,26 @@ const useCreateConceptDialog = workspaceId => {
...
       required: true
     }, {
+      type: 'textfield',
       name: 'description',
       multiline: true
+    }, {
+      type: 'select',
+      nullable: true,
+      name: 'tags',
+      label: 'Select Bloom\'s tags',
```
Comment (tulir) [inline frontend/src/dialogs/concept/useCreateConceptDialog.js]:
```suggestion
      label: "Select Bloom's tags",
```

File: frontend/src/dialogs/concept/useEditConceptDialog.js [hunk 4]
```diff
@@ -21,13 +21,29 @@ const useEditConceptDialog = () => {
...
       defaultValue: name
     }, {
+      type: 'textfield',
       name: 'description',
       multiline: true,
       defaultValue: description
+    }, {
+      type: 'select',
+      name: 'tags',
+      label: 'Select Bloom\'s tags',
```
Comment (tulir) [inline frontend/src/dialogs/concept/useEditConceptDialog.js]:
```suggestion
      label: "Select Bloom's tags",
```

File: frontend/src/dialogs/concept/useEditConceptDialog.js [hunk 5]
```diff
@@ -21,13 +21,29 @@ const useEditConceptDialog = () => {
...
+      nullable: true,
+      defaultValue: tags[0].name,
+      values: [
+        'REMEMBER',
+        'UNDERSTAND',
+        'APPLY',
+        'ANALYZE',
+        'EVALUATE',
+        'CREATE'
+      ]
```
Comment (tulir) [inline frontend/src/dialogs/concept/useEditConceptDialog.js]: these should probably be in a shared `const` somewhere

File: frontend/src/dialogs/concept/useEditConceptDialog.js [hunk 6]
```diff
@@ -21,13 +21,22 @@ const useEditConceptDialog = () => {
...
+      type: 'textfield',
       name: 'description',
       multiline: true,
       defaultValue: description
+    }, {
+      type: 'select',
+      name: 'tags',
+      label: "Select Bloom's tags",
+      isList: true,
+      defaultValue: tags.length > 0 ? tags[0].name : '',
```
Comment (tulir) [inline frontend/src/dialogs/concept/useEditConceptDialog.js]:
```suggestion
      defaultValue: tags.find(tag => tag.type === 'bloom') || '',
```

File: frontend/src/dialogs/concept/useEditConceptDialog.js [hunk 7]
```diff
@@ -21,13 +21,22 @@ const useEditConceptDialog = () => {
...
     }, {
+      type: 'textfield',
       name: 'description',
       multiline: true,
       defaultValue: description
+    }, {
+      type: 'select',
+      name: 'tags',
+      label: "Select Bloom's tags",
+      isList: true,
```
Comment (tulir) [inline frontend/src/dialogs/concept/useEditConceptDialog.js]: there's only one of bloom's tags at a time though :thinking:

**Pull request merged**